src/main.ts

```diff
@@ -10,43 +10,50 @@ import { SeedingService } from './database/seeding/seeding.service';
 import { ResponseInterceptor } from './shared/inteceptors/response.interceptor';
 
 async function bootstrap() {
-  const app = await NestFactory.create<NestExpressApplication>(AppModule, { bufferLogs: true });
+  try {
+    const app = await NestFactory.create<NestExpressApplication>(AppModule, { bufferLogs: true });
 
-  const logger = app.get(Logger);
+    const logger = app.get(Logger);
 
-  const dataSource = app.get(DataSource);
+    const dataSource = app.get(DataSource);
 
-  try {
-    await initializeDataSource();
-    console.log('Data Source has been initialized!');
-  } catch (err) {
-    console.error('Error during Data Source initialization', err);
-    process.exit(1);
-  }
+    try {
+      await initializeDataSource();
+      console.log('Data Source has been initialized!');
+    } catch (err) {
+      console.error('Error during Data Source initialization', err);
+      process.exit(1);
+    }
 
-  const seedingService = app.get(SeedingService);
-  await seedingService.seedDatabase();
+    const seedingService = app.get(SeedingService);
+    await seedingService.seedDatabase();
 
-  app.enable('trust proxy');
-  app.useLogger(logger);
-  app.enableCors();
-  app.setGlobalPrefix('api/v1', { exclude: ['/', 'health'] });
-  app.useGlobalInterceptors(new ResponseInterceptor());
+    app.enable('trust proxy');
+    app.useLogger(logger);
+    app.enableCors();
+    app.setGlobalPrefix('api/v1', { exclude: ['/', 'health', 'api', 'api/v1', 'api/docs'] });
 
-  // TODO: set options for swagger docs
-  const options = new DocumentBuilder()
-    .setTitle('<project-title-here>')
-    .setDescription('<project-description-here>')
-    .setVersion('1.0')
-    .addBearerAuth()
-    .build();
+    // TODO: set options for swagger docs
+    const options = new DocumentBuilder()
+      .setTitle('<project-title-here>')
+      .setDescription('<project-description-here>')
+      .setVersion('1.0')
+      .addBearerAuth()
+      .build();
 
-  const document = SwaggerModule.createDocument(app, options);
-  SwaggerModule.setup('api', app, document);
+    const document = SwaggerModule.createDocument(app, options);
+    SwaggerModule.setup('api/docs', app, document);
 
-  const port = app.get<ConfigService>(ConfigService).get<number>('server.port');
-  await app.listen(port);
+    const port = app.get<ConfigService>(ConfigService).get<number>('server.port');
+    await app.listen(port);
 
-  logger.log({ message: 'server started 🚀', port, url: `http://localhost:${port}/api` });
+    logger.log({ message: 'server started 🚀', port, url: `http://localhost:${port}/api/v1` });
+  } catch (err) {
+    console.error('Error during bootstrap', err);
+    process.exit(1);
+  }
 }
-bootstrap();
+bootstrap().catch(err => {
+  console.error('Error during bootstrap', err);
+  process.exit(1);
+});
```

Comment:
This change wasn't necessary

Reply:
How so?

Reply:
![image](https://github.com/user-attachments/assets/adcb46c4-3622-4a1f-a915-3b4f6a7138b1)